src/rdysetup.c

```diff
@@ -793,9 +792,9 @@ static PetscErrorCode InitSolution(RDy rdy) {
           for (PetscInt c = 0; c < region.num_cells; ++c) {
             PetscInt cell_id = region.cell_ids[c];
             if (ndof * cell_id < n_local) {  // skip ghost cells
-              x_ptr[3 * cell_id]     = mupEval(flow_ic.height);
-              x_ptr[3 * cell_id + 1] = mupEval(flow_ic.x_momentum);
-              x_ptr[3 * cell_id + 2] = mupEval(flow_ic.y_momentum);
+              u_ptr[3 * cell_id]     = mupEval(flow_ic.height);
+              u_ptr[3 * cell_id + 1] = mupEval(flow_ic.x_momentum);
+              u_ptr[3 * cell_id + 2] = mupEval(flow_ic.y_momentum);
```

Comment:
Since we are editing these lines, how about changing `u_ptr[3 * ...` to `u_ptr[ndof * ...`?